### PR TITLE
Add `rstuf task info` command

### LINKED ISSUE #335: Task: Add command to retrieve task status `rstuf task info`
### What is the task about?

This feature is about adding a command to retrieve the task status.

Note: The `-t/--token` is ignored if the user uses `rstuf --auth`

`rstuf task info <task-id> [-t/--token <token>]`

Print the JSON using the print json formatted from the click.

### Parent feature

- #334 

### References

_No response_

### Depends on
- https://github.com/repository-service-tuf/repository-service-tuf-worker/issues/315
- https://github.com/repository-service-tuf/repository-service-tuf-cli/issues/339

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by KAUTH**
# Description

Closes #335

# Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)

# Additional requirements
- [x] Tests have been added for the bug fix or new feature
- [x] Docs have been added for the bug fix or new feature

# Code of Conduct
By submitting this PR, you agree to follow our [Code of Conduct](CODE_OF_CONDUCT.rst).

- [x] I agree to follow this project's Code of Conduct

### CODE REVIEW

**Review by MVrachev — COMMENTED** [review]
I tried running `rstuf task info` without providing any server or `api-server` argument 
and I got the error `AttributeError: 'Settings' object has no attribute 'SERVER'`.
I am wondering can't we somehow give a better documentation how do you setup
this attribute `SERVER` in the settings?

Then I realized there is no `api-server` option for this command.
Please add such an option, so I don't have to be forced to log in or do anything as a pre-requirement.

On another note: do you know how we generate `rstuf.yml`? 
As I don't see a command to create one do we expect our users to create it manually?

**Comment by KAUTH** [comment]
> Please add such an option, so I don't have to be forced to log in or do anything as a pre-requirement.

👍

> On another note: do you know how we generate rstuf.yml?
As I don't see a command to create one do we expect our users to create it manually?

That is a very good point. I have also looked and there's no documentation for `rstuf.yml`.

File: repository_service_tuf/cli/task/info.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    """
+
+    settings = context.obj.get("settings")
+
+    if api_server:
+        settings.SERVER = api_server
+
+    if settings.get("SERVER") is None:
+        raise click.ClickException(
+            "Requires '--api-server' or configuring the `.rstuf.yml` file. "
```
Comment (MVrachev) [inline repository_service_tuf/cli/task/info.py:38]:
```suggestion
            "Requires '--api-server' or configuring the `rstuf.yml` file. "
```

File: tests/unit/cli/task/test_info.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+        task_id = "0123456789"
+
+        input_steps = [
+            task_id,
+        ]
+
+        result = client.invoke(info.info, input_steps, obj=test_context)
+        output_message = (
+            "Requires '--api-server' or configuring the `.rstuf.yml` file."
```
Comment (MVrachev) [inline tests/unit/cli/task/test_info.py:53]:
```suggestion
            "Requires '--api-server' or configuring the `rstuf.yml` file."
```

**Review by MVrachev — CHANGES_REQUESTED** [review]
Two nits remaining.

File: repository_service_tuf/cli/task/info.py (hunk 1 above)
Comment (KAUTH) [reply repository_service_tuf/cli/task/info.py:38]: I believe the config file is a dotfile, see https://github.com/repository-service-tuf/repository-service-tuf-cli/blob/main/repository_service_tuf/cli/__init__.py#L46 and https://github.com/repository-service-tuf/repository-service-tuf-cli/blob/main/docs/source/guide/index.rst.

Comment (MVrachev) [reply repository_service_tuf/cli/task/info.py:38]: You have a point.

**Review by MVrachev — APPROVED** [review]

**Review by MVrachev — APPROVED** [review]

**Pull request merged**